atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CqlExecutor.java

```diff
@@ -60,66 +62,61 @@
-        return query.executeAndGetCells(row);
+        return query.executeAndGetCells(startRowInclusive, this::getCellForKeyColumn1Column2);
     }
 
     /**
-     * @param tableRef the table from which to select
-     * @param row the row key
-     * @param column the column name
-     * @param maxTimestampExclusive the maximum timestamp, exclusive
-     * @param limit the maximum number of results to return.
-     * @return up to <code>limit</code> cells that exactly match the row and column name, and have a timestamp less than
-     * <code>maxTimestampExclusive</code>
+     * Returns a list of {@link CellWithTimestamp}s within the given {@code row}, starting at the (column, timestamp)
+     * pair represented by ({@code startColumnInclusive}, {@code startTimestampExclusive}).
      */
-    List<CellWithTimestamp> getTimestampsForRowAndColumn(
+    public List<CellWithTimestamp> getTimestampsWithinRow(
             TableReference tableRef,
             byte[] row,
-            byte[] column,
-            long maxTimestampExclusive,
+            byte[] startColumnInclusive,
+            long startTimestampExclusive,
             int limit) {
-        long invertedTimestamp = ~maxTimestampExclusive;
+        long invertedTimestamp = ~startTimestampExclusive;
         CqlQuery query = new CqlQuery(
-                "SELECT column1, column2 FROM %s WHERE key = %s AND column1 = %s AND column2 > %s LIMIT %s;",
+                "SELECT column1, column2 FROM %s WHERE key = %s AND (column1, column2) > (%s, %s) LIMIT %s;",
                 quotedTableName(tableRef),
                 key(row),
-                column1(column),
+                column1(startColumnInclusive),
                 column2(invertedTimestamp),
                 limit(limit));
-        return query.executeAndGetCells(row);
+        return query.executeAndGetCells(row, result -> getCellForColumn1Column2(result, row));
     }
 
-    /**
-     * @param tableRef the table from which to select
-     * @param row the row key
-     * @param previousColumn the lexicographic lower bound (exclusive) for the column name
-     * @param limit the maximum number of results to return.
-     * @return up to <code>limit</code> results where the column name is lexicographically later than the supplied
-     * <code>previousColumn</code>. Note that this can return results from multiple columns
-     */
-    List<CellWithTimestamp> getNextColumnsForRow(
-            TableReference tableRef,
-            byte[] row,
-            byte[] previousColumn,
-            int limit) {
-        CqlQuery query = new CqlQuery(
-                "SELECT column1, column2 FROM %s WHERE key = %s AND column1 > %s LIMIT %s;",
-                quotedTableName(tableRef),
-                key(row),
-                column1(previousColumn),
-                limit(limit));
-        return query.executeAndGetCells(row);
+    private CellWithTimestamp getCellForColumn1Column2(CqlRow row, byte[] key) {
```

Comment:
nit: `getCellFromColumn1Column2`?

The `Column1Column2` here looks a bit nasty, but I struggle to think of better wording. Perhaps `getCellFromKeylessRow`?